File: Java/src/commando/AbstractCommand.java
package commando;

public abstract class AbstractCommand {

    public String execute(String command) throws InvalidCommandException {
        String[] parts = command.split(",");
        try {
            long firstNumber = Long.parseLong(parts[0]);
            long secondNumber = Long.parseLong(parts[1]);
            return String.valueOf(doSomethingWith(firstNumber, secondNumber));
        } catch (IllegalArgumentException e) {
            throw new InvalidCommandException(command, e);
        }
    }

    protected abstract long doSomethingWith(long firstNumber, long secondNumber);
}

File: PHP/src/Commando/AbstractCommand.php
<?php
namespace Commando;

abstract class AbstractCommand {

    public function execute($command) {
        $parts = explode(",", $command);
        $firstNumber = (int)$parts[0];
        if ($parts[0] !== (string)$firstNumber) {
            throw new InvalidCommandException($command);
        }
        $secondNumber = (int)$parts[1];
        if ($parts[1] !== (string)$secondNumber) {
            throw new InvalidCommandException($command);
        }
        return strval($this->doSomethingWith($firstNumber, $secondNumber));
    }

    protected abstract function doSomethingWith($firstNumber, $secondNumber);
}
